aten/src/ATen/test/math_kernel_test.cpp

```diff
@@ -0,0 +1,40 @@
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+
+using namespace at;
+
+#define ASSERT_ALLCLOSE_TOLERANCES(t1, t2, atol, rtol) \
+  ASSERT_TRUE(t1.is_same_size(t2));                    \
+  ASSERT_TRUE(t1.allclose(t2, atol, rtol));
+
+// Ideally we want to test both forward and backward on math kernels but I
+// haven't found an easy way to do it.  Currently we only test forward here
+// and rely on backward tests of each at:: function used in math kernels.
```

Comment:
This is definitely something we should work on making easier, though I'm not exactly sure how. Perhaps APIs for supporting static runtime would make it easier, as they would make it possible to ask for a specific kernel.

Absent machinery like this, I think the easiest way to do this is to just make the math kernel available at another name (i.e., make a new `native_functions.yaml` entry; and then just do the testing from Python. Maybe @mruberry has some other ideas.

Reply:
Without too much context I'll also vote for (re)using Python.